bonsai-core/src/main/scala/com/stripe/bonsai/FullBinaryTreeOps.scala

```diff
@@ -21,6 +21,17 @@ trait FullBinaryTreeOps[T, BL, LL] extends TreeOps[T, Either[BL, LL]] {
 
   def children(node: Node): Iterable[Node] =
     foldNode(node)({ case (lc, rc, _) => lc :: rc :: Nil }, _ => Nil)
+
+  def collectLabelsF[A](node: Node, f: Label => A): Set[A] =
```

Comment:
If you wanted to, you could implement this w/ `reduce`:

`reduce(node) { case (label, childLabels) => childLabels.foldLeft(Set(label))(_ ++ _) }`

Though I'd say there is a case to be made that we should have a `FullBinaryTreeOps`-specific version of reduce that takes 2 functions, since we shouldn't have to deal with the `Iterable` `childLabels` (and it'd make implementing `collectLeafLabels` in terms of reduce nicer):

```scala
def reduce[A](node: Node)(f: LL => A)(g: (BL, A, A) => A): A = ???
```

wdyt?